ksql-serde/src/main/java/io/confluent/ksql/serde/json/KsqlJsonDeserializer.java

```diff
@@ -163,34 +164,31 @@ private static Struct enforceFieldTypesForStruct(final JsonValueContext context)
     }
 
     final Struct columnStruct = new Struct(context.schema);
-    final Map<String, ?> fields =
-        toCaseInsensitiveFieldNameMap((Map<String, ?>) context.val, context.topLevel);
-
-    context.schema.fields().forEach(
-        field -> {
-          final Object fieldValue = fields.get(field.name().toUpperCase());
-          final Object coerced = enforceFieldType(
-              context.deserializer, field.schema(), fieldValue, false);
-          columnStruct.put(field.name(), coerced);
-        });
+    final Map<String, ?> fields = (Map<String, ?>) context.val;
```

Comment:
nit: since we're dealing with both ksql and json fields here, can we use a prefix to identify which a variable refers to. E.g. here, we're fields could be `jsonFields`. And in the loop below: `for (Field ksqlField : context.schema.fields()) {`